### PR TITLE
fix(api_client): limit in-flight requests

### PR DESCRIPTION
**Pull request by opichals**
Limit the maximum number of fetch_all page requests running at in parallel.

### CODE REVIEW

File: rossum_api/api_client.py [hunk 1]
```diff
@@ -175,23 +188,49 @@ async def fetch_all(
...
+                    json=json,
                 )
-            )
-            for i in range(2, last_page + 1)
+
+            return fetch_page
+
+        # initiate max_in_flight_requests requests in parallel
+        pending_request_tasks = [
+            *self._run_tasks(_fetch_page_request, self.max_in_flight_requests),
```
Comment (asgeirrr) [inline rossum_api/api_client.py:209]: This code feels quite complex. Looks like `asyncio` has a primitive called [asyncio.Semaphore](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore) for this. Can we use it? :pray: There might be smt even better but this is what I found :wink:

**Review by asgeirrr — CHANGES_REQUESTED** [review]
Thanks, this is a nice addition in order not to DoS Elis :laughing:

File: rossum_api/api_client.py (hunk 1 above)
Comment (opichals) [reply rossum_api/api_client.py]: @asgeirrr Refactored to use `asyncio.Semaphore`. Could you have another look?

File: rossum_api/api_client.py [hunk 2]
```diff
@@ -175,14 +177,20 @@ async def fetch_all(
...
-            asyncio.create_task(
-                self._fetch_page(
-                    f"{resource}", method, {**query_params, "page": i}, sideloads, json=json
+
+        in_flight_guard = asyncio.Semaphore(self.max_in_flight_requests)
+
+        async def _fetch_page(page_number):
+            async with in_flight_guard:
+                return await self._fetch_page(
+                    f"{resource}",
```
Comment (robinoburka) [inline rossum_api/api_client.py:186]: `f"{resource}"` is the same as `resource` in this case. But I accept it was in the previous code. Up to you.

**Review by robinoburka — APPROVED** [review]
Small nitpick. Up to you. Solution works for me. Approved.

**Review by asgeirrr — APPROVED** [review]
This is pure gold, thanks a lot :1st_place_medal:  I like the thorough test.

File: rossum_api/api_client.py [hunk 3]
```diff
@@ -106,6 +106,7 @@ def __init__(
         timeout: Optional[float] = None,
         n_retries: int = 3,
         retry_backoff_factor: float = 1.0,
+        max_in_flight_requests: int = 4,
```
Comment (asgeirrr) [inline rossum_api/api_client.py:109]: This option should probably be passable from the user-facing classes `ElisAPIClient(Sync)` to allow users to customize it easily but let's merge this as is.

**Pull request merged**